File: backend/app/schemas/caso_noti.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class CasoNotiBase(BaseModel):
    id_caso_noti: str
    departamento: str
    provincia: str
    distrito: str
    ubigeo: str
    enfermedad: str
    ano_epidemiologico: int
    semana_epidemiologica: int
    tipo_diagnostico: str
    diresa_notifica: str
    edad: int
    sexo: str


class CasoNotiResponse(CasoNotiBase):
    id: uuid.UUID
    fecha_creacion: datetime

    model_config = {"from_attributes": True}

File: dashboard/src/lib/schemas/casoNoti.ts
import { z } from 'zod'

// Refleja los campos del modelo CasoNoti del backend
export const CasoNotiSchema = z.object({
  id: z.string().uuid(),
  id_caso_noti: z.string().max(100),

  // Ubicación administrativa
  departamento: z.string().max(100),
  provincia: z.string().max(100),
  distrito: z.string().max(100),
  ubigeo: z.string().max(10),

  // Datos epidemiológicos
  enfermedad: z.string().max(150),
  ano_epidemiologico: z.number().int().min(2000).max(2100),
  semana_epidemiologica: z.number().int().min(1).max(53),
  // 'C'=confirmado, 'P'=probable, 'S'=sospechoso
  tipo_diagnostico: z.enum(['C', 'P', 'S']),
  diresa_notifica: z.string().max(150),

  // Datos del paciente
  edad: z.number().int().min(0).max(120),
  sexo: z.enum(['M', 'F']),
})

export type CasoNoti = z.infer<typeof CasoNotiSchema>

export const CasoNotiListSchema = z.array(CasoNotiSchema)
